logstash-core/src/main/java/org/logstash/common/clients/HttpClient.java

```diff
@@ -0,0 +1,701 @@
+                    }
+                } else {
+                    trustManagers = new TrustManager[] { getAnyTrustManager() };
+                }
+
+                KeyManager[] keyManagers = null;
+                if ((this.sslClientCertificate != null) && (this.sslClientPrivateKey != null)) {
+                    KeyStore keystore = getKeyStore();
+
+                    try {
+                        keystore.setKeyEntry("clientPrivateKey", this.sslClientPrivateKey, "".toCharArray(), new Certificate[]{this.sslClientCertificate});
+
+                        KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+                        keyManagerFactory.init(keystore, "".toCharArray());
+
+                        keyManagers = keyManagerFactory.getKeyManagers();
+                    } catch (Exception e) {
+                        throw new OptionsBuilderException("Unable to use provided client certificate and/or client private key", e);
+                    }
+                }
+
+                try {
+                    SSLContext sslContext = SSLContext.getInstance("TLS");
+                    sslContext.init(keyManagers, trustManagers, null);
+
+                    SSLConnectionSocketFactory sslSocketFactory;
+                    if (this.sslVerifyServerHostname) {
+                        sslSocketFactory = new SSLConnectionSocketFactory(sslContext, SSLConnectionSocketFactory.STRICT_HOSTNAME_VERIFIER);
+                    } else {
+                        sslSocketFactory = new SSLConnectionSocketFactory(sslContext, SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
+                    }
+
+                    httpClientBuilder.setSSLSocketFactory(sslSocketFactory);
+                } catch (Exception e) {
+                    throw new OptionsBuilderException("Unable to configure HTTP client with SSL/TLS", e);
+                }
+            }
+
+            if (usesBasicAuth()) {
+                String credentials = this.basicAuthUsername + ":" + this.basicAuthPassword;
+                String encodedCredentials = new String(Base64.getEncoder().encode(credentials.getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8);
+
+                List<Header> headerList = Collections.singletonList(new BasicHeader("Authorization", "Basic " + encodedCredentials));
+                httpClientBuilder.setDefaultHeaders(headerList);
+            }
+
+            return new HttpClient(httpClientBuilder.build(), baseUrl);
+        }
+
+        private boolean usesBasicAuth() {
+            return (this.basicAuthUsername != null) && (this.basicAuthPassword != null);
+        }
+
+        private boolean usesSsl(URL baseUrl) {
+            return baseUrl.getProtocol().equals("https");
+        }
+
+        private static X509Certificate getCertificate(String certificateFilePath) throws OptionsBuilderException {
+            FileInputStream fis;
```

Comment:
use `try-with-resources` on the `fis` here imo :) Much shorter and easier to read